src/rest.ts

```diff
@@ -0,0 +1,100 @@
+                    case ('/api/port'): {
+                        res.writeHead(200, { "Content-Type": "application/json" })
+                        res.write(String(this._cs_client.port))
+                        res.end()
+                        break
+                    }
+                    default: {
+                        res.writeHead(204, { "Content-Type": "application/json" })
+                        res.end(JSON.stringify({ message: "Route not found" }))
+                        break
+                    }
+                }
+            }
+            if (req.method === 'POST') {
+                let body = ''
+                req.on('data', (data) => {
+                    body += data
+                })
+                req.on('end', () => {
+                    console.log("api post:", req.url, body)
+                    switch (req.url) {
+                        case ('/api/ip'): {
+                            if (net.isIP(body) != 0) {
+                                res.writeHead(200, { "Content-Type": "application/json" })
+                                res.end()
+                                this._cs_client.connect(body, this._cs_client.port)
+                            } else {
+                                res.writeHead(400, { "Content-Type": "application/json" })
+                                res.end(JSON.stringify({ message: "Malformed ip" }))
+                            }
+                            break
+                        }
+                        case ('/api/port'): {
+                            let port = Number(body)
+                            if (port && port > 0 && port < 65535) {
+                                res.writeHead(200, { "Content-Type": "application/json" })
+                                res.end()
+                                this._cs_client.connect(this._cs_client.host, port)
+                            } else {
+                                res.writeHead(400, { "Content-Type": "application/json" })
+                                res.end(JSON.stringify({ message: "bad port" }))
+                            }
+                            break
+                        }
+                        default: {
+                            res.writeHead(204, { "Content-Type": "application/json" })
+                            res.end(JSON.stringify({ message: "Route not found" }))
+                            break
+                        }
+                    }
+                })
+            }
+        })
+
+        this.open(port)
+    }
+
+    public open(port: Number) {
+        if (this._server.listening) { this.close() }
+        if (port != 0) {
+            this._server.listen(port, () => {
+                console.log(`REST server starting: port: ${port}`)
+            })
+        } else {
+            console.log("REST server not starting: port 0")
+        }
+
+    }
+
+    public close(): void {
+        this._server.closeAllConnections()
```

Comment:
```suggestion
        this._server.closeAllConnections()
        this._server.close()
```
To stop listening for new connections too